docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,255 @@
+---
+id: 'effective-gno'
+---
+
+# Effective Gno
+
+Welcome to the guide for writing effective Gno code. This document is designed
+to help you understand the nuances of Gno and how to use it effectively.
+
+Before we dive in, it's important to note that Gno shares several similarities
+with Go. Therefore, if you haven't already, we highly recommend reading
+["Effective Go"](https://go.dev/doc/effective_go) as a primer.
+
+## Counter-Intuitive Good Practices
+
+This section highlights some Gno good practices that might seem
+counter-intuitive, especially if you're coming from a Go background.
+
+### Embrace Global Variables in Gno
+
+In Gno, using global variables isn't just acceptable - it's encouraged. This is
+because global variables in Gno provide a way to have persisted states
+automatically.
+
+In Go, you would typically write your logic and maintain some state in memory.
+However, to persist the state and ensure it survives a restart, you would need
+to use a store (like a plain file, custom file structure, a database, a
+key-value store, an API, etc.).
+
+In contrast, Gno simplifies this process. When you declare global variables in
+Gno, the GnoVM automatically persists and restores them as needed between each
+run.
+
+However, be mindful not to export your global variables. Doing so would make
+them accessible for everyone to read and write.
+
+Here's an ideal pattern to follow:
+
+```go
+var counter int
+
+func GetCounter() int {
+    return counter
+}
+
+func IncCounter() {
+    counter++
+}
+```
+
+### Embrace Panic in Gno
+
+In Gno, it's important to know when to return an `error` and when to use `panic()`.
+Each does something different to your code and data.
+
+When you return an error in Gno, it's like giving back any other piece of data.
+It tells you something went wrong, but it doesn't stop your code or undo any
+changes you made.
+
+But, when you use panic in Gno, it stops your code right away, says it failed,
+and doesn't save any changes you made. This is safer when you want to stop
```

Comment:
Consider adding the keyword `revert` for those coming from other chain-specific languages who are familiar with similar behavior

Reply:
I suggest the following:
1. Create a specific discussion topic for considering the addition of the keyword.
2. Find a suitable location in the documentation to discuss this, perhaps "from-solidity-to-gno.md" instead of effective-gno.

Reply:
I believe you're referring to the keyword, not for the gno language, but rather the keyword in the context of the document, such as for "SEO" or searchability purposes. 

Certainly, can you suggest an alternative?

Reply:
Ah yes, not so much a reserved keyword but more as SEO keyword. As I understand it, our version of `revert` (Solidity) is essentially, `panic`.

Reply:
Good point, perhaps better for Solidity-specific reference than general Effective Gno. Will resolve.